contracts/StableJoeStaking.sol

```diff
@@ -0,0 +1,270 @@
+     * @param _amount The amount of JOE to withdraw
+     */
+    function withdraw(uint256 _amount) external {
+        UserInfo storage user = userInfo[msg.sender];
+        require(user.amount >= _amount, "StableJoeStaking: withdraw amount exceeds balance");
+
+        updatePool();
+        uint256 pending = user.amount.mul(accTokenPerShare).div(PRECISION).sub(user.rewardDebt);
+
+        user.amount = user.amount.sub(_amount);
+        user.rewardDebt = user.amount.mul(accTokenPerShare).div(PRECISION);
+
+        safeTokenTransfer(msg.sender, pending);
+        joe.safeTransfer(address(msg.sender), _amount);
+        emit Withdraw(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Withdraw without caring about rewards. EMERGENCY ONLY
+     */
+    function emergencyWithdraw() external {
+        UserInfo storage user = userInfo[msg.sender];
+
+        uint256 _amount = user.amount;
+        user.amount = 0;
+        user.rewardDebt = 0;
+
+        joe.safeTransfer(address(msg.sender), _amount);
+        emit EmergencyWithdraw(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Update reward variables to be up-to-date
+     * @dev Needs to be called before any deposit or withdrawal
+     */
+    function updatePool() public {
+        if (block.timestamp <= lastRewardTimestamp) {
+            return;
+        }
+
+        uint256 lpSupply = joe.balanceOf(address(this));
+        if (lpSupply == 0) {
+            lastRewardTimestamp = block.timestamp;
+            return;
+        }
+
+        uint256 multiplier;
+        uint256 tokenReward;
+        // update values to be accurate with today's buyback
+        if (block.timestamp > lastDailyFeeTimestamp + 1 days) {
+            // remaining time of the previous day
+            multiplier = 1 days - (lastRewardTimestamp % 1 days);
+
+            tokenReward = multiplier.mul(tokensPerSecScaled);
+
+            // now we update the values for the new day
+            lastRewardTimestamp = lastDailyFeeTimestamp + 1 days;
+            // get the current day at 00:00:00, _lastRewardTimestamp and _lastDailyFeeTimestamp
+            // can be different, if `updatePool` isn't called at least once per day.
+            lastDailyFeeTimestamp = (block.timestamp / 1 days) * 1 days;
+
+            // get today's buyback amount
+            uint256 accruedFee = rewardToken.balanceOf(address(this)).sub(lastTokenBalance);
+            lastTokenBalance = lastTokenBalance.add(accruedFee);
+
+            // distribute today's buyback over 1 day
+            tokensPerSecScaled = accruedFee.mul(PRECISION).div(1 days);
+        }
+
+        // in case `updatePool` isn't called in more than 2 days, we need to make sure user receives only one day
+        // worth of token
+        if (block.timestamp > lastRewardTimestamp + 1 days) {
+            tokenReward = tokenReward.add(uint256(1 days).mul(tokensPerSecScaled));
+            tokensPerSecScaled = 0;
```

Comment:
As we assume that a buyback happens every day, and therefore that updatePool is called at least once a day. This will never be true.

But if for what ever reason, `updatePool` isn't called for one full day (see coments above), then we still need to distribute the rewards that were accrued, but it only takes into account one day of reward.
So if we're not updating this to 0, the previous rewards will keep being distributed and we will be in debt.
After that if a buyback happen, it will be distributed the next day.

So:
day 0:
tokenPerSec = 0
accruedRewards = 86400

day 1:
tokenPerSec = 1
accruedRewards = 0

day 2:
tokenPerSec = 0
accruedRewards = X


if X is greater than 0, then rewards will be distributed day 3, if not it will wait until accruedRewards > X in the upcoming days, and will distribute them the next day